provider/identityprovider/identity_provider_resource.go

```diff
@@ -23,108 +23,122 @@ import (
+			"htpasswd": schema.SingleNestedAttribute{
 				Description: "Details of the 'htpasswd' identity provider.",
-				Attributes:  idps.HtpasswdSchema(),
+				Attributes:  htpasswdSchema,
 				Optional:    true,
-				Validators:  idps.HTPasswdValidators(),
 			},
-			"gitlab": {
+			"gitlab": schema.SingleNestedAttribute{
 				Description: "Details of the Gitlab identity provider.",
-				Attributes:  idps.GitlabSchema(),
+				Attributes:  gitlabSchema,
 				Optional:    true,
-				Validators:  idps.GitlabValidators(),
+				Validators:  GitlabValidators(),
 			},
-			"github": {
+			"github": schema.SingleNestedAttribute{
 				Description: "Details of the Github identity provider.",
-				Attributes:  idps.GithubSchema(),
+				Attributes:  githubSchema,
 				Optional:    true,
-				Validators:  idps.GithubValidators(),
+				Validators:  GitlabValidators(),
 			},
-			"google": {
+			"google": schema.SingleNestedAttribute{
 				Description: "Details of the Google identity provider.",
-				Attributes:  idps.GoogleSchema(),
+				Attributes:  googleSchema,
 				Optional:    true,
-				Validators:  idps.GoogleValidators(),
+				Validators:  GoogleValidators(),
 			},
-			"ldap": {
+			"ldap": schema.SingleNestedAttribute{
 				Description: "Details of the LDAP identity provider.",
-				Attributes:  idps.LDAPSchema(),
+				Attributes:  ldapSchema,
 				Optional:    true,
-				Validators:  idps.LDAPValidators(),
+				Validators:  LDAPValidators(),
 			},
-			"openid": {
+			"openid": schema.SingleNestedAttribute{
 				Description: "Details of the OpenID identity provider.",
-				Attributes:  idps.OpenidSchema(),
+				Attributes:  openidSchema,
 				Optional:    true,
 			},
 		},
 	}
 	return
 }
 
-func (t *IdentityProviderResourceType) NewResource(ctx context.Context,
-	p tfsdk.Provider) (result tfsdk.Resource, diags diag.Diagnostics) {
-	// Cast the provider interface to the specific implementation:
-	// use it directly when needed.
-	parent := p.(*Provider)
+func (r *IdentityProviderResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	collection, ok := req.ProviderData.(*sdk.Connection)
+	if !ok {
+		resp.Diagnostics.AddError("Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *sdk.Connaction, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.collection = collection.ClustersMgmt().V1().Clusters()
+}
 
-	// Get the collection of clusters:
-	collection := parent.connection.ClustersMgmt().V1().Clusters()
+func (r *IdentityProviderResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	idp := &IdentityProviderState{}
+	diag := req.Config.Get(ctx, idp)
 
-	// Create the resource:
-	result = &IdentityProviderResource{
-		collection: collection,
+	if diag.HasError() {
+		return
 	}
 
-	return
+	//TODO: add validations
```

Comment:
Please add the validation

Reply:
I am uncertain if we need in this level. We have validations for all the rest. The only things without validations are `cluster` `id` and `name`...  Ill remove this function

Reply:
On second thought, maybe we want to be sure just one of the idp "providers/methods" are passed. (this validation was not present in the previous implementation.

Reply:
I don't want to hold up this PR, as the main focus is to get the code ported over.
This resource has many validations in it, and I think a fair number of them can be converted to use some of the framework's built-in validators.
Examples:
- htpassword username and password checks can use `stringvalidator.RegexMatches()` and the min/max length validators.
- The mappingmethod validator can be a `stringvalidator.OneOf()`

I suspect we can also cut a fair bit of logic in this and other resources by adding ConflictsWith() and/or ExactlyOneOf()

Reply:
Added a ticket https://issues.redhat.com/browse/OCM-3735